ecosystem/typescript/sdk/src/token_client.ts

```diff
@@ -47,18 +43,31 @@ export class TokenClient {
     description: string,
     uri: string,
   ): Promise<Types.HexEncodedBytes> {
-    const payload: Types.TransactionPayload = {
-      type: 'script_function_payload',
-      function: '0x1::Token::create_unlimited_collection_script',
-      type_arguments: [],
-      arguments: [
-        Buffer.from(name).toString('hex'),
-        Buffer.from(description).toString('hex'),
-        Buffer.from(uri).toString('hex'),
-      ],
-    };
-    const transactionHash = await this.submitTransactionHelper(account, payload);
-    return transactionHash;
+    const payload = new TxnBuilderTypes.TransactionPayloadScriptFunction(
+      TxnBuilderTypes.ScriptFunction.natural(
+        '0x1::Token',
+        'create_unlimited_collection_script',
+        [],
+        [BCS.bcsSerializeStr(name), BCS.bcsSerializeStr(description), BCS.bcsSerializeStr(uri)],
+      ),
+    );
+
+    const [{ sequence_number: sequnceNumber }, chainId] = await Promise.all([
+      this.aptosClient.getAccount(account.address()),
+      this.aptosClient.getChainId(),
+    ]);
+
+    const rawTxn = new TxnBuilderTypes.RawTransaction(
+      TxnBuilderTypes.AccountAddress.fromHex(account.address()),
+      BigInt(sequnceNumber),
+      payload,
+      1000n,
```

Comment:
just curious if the max_gas amount and gas unit price provided as argument from users? or we use default value here just to provide an ergonomic API?

Reply:
Yeah in the past I have needed to raise the max gas limit when working on my dapps, these should probably be kwargs with defaults.

Reply:
This is a drop-in replacement. Behind the scenes, the JSON txn also used the same gas units and price. Till we do next major version release, let's just keep the interface unchanged.